app/uk/gov/hmrc/alcoholdutyaccount/service/AlcoholDutyService.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.alcoholdutyaccount.service
+
+import cats.data.EitherT
+import play.api.http.Status.{NOT_FOUND, NOT_IMPLEMENTED}
+import uk.gov.hmrc.alcoholdutyaccount.connectors.{FinancialDataConnector, ObligationDataConnector, SubscriptionSummaryConnector}
+import uk.gov.hmrc.alcoholdutyaccount.models.{AlcoholDutyCardData, Approved, ErrorResponse, InsolventCardData, Payments, Returns, hods}
+import uk.gov.hmrc.alcoholdutyaccount.models.hods.{Document, ObligationDetails, Open, SubscriptionSummary}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import java.time.LocalDate
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton()
+class AlcoholDutyService @Inject() (
+  subscriptionSummaryConnector: SubscriptionSummaryConnector,
+  obligationDataConnector: ObligationDataConnector,
+  financialDataConnector: FinancialDataConnector
+)(implicit ec: ExecutionContext) {
+
+  def getAlcoholDutyCardData(
+    alcoholDutyReference: String
+  )(implicit hc: HeaderCarrier): EitherT[Future, ErrorResponse, AlcoholDutyCardData] = EitherT {
+    subscriptionSummaryConnector.getSubscriptionSummary(alcoholDutyReference).value.flatMap {
+      case None                      => Future.successful(Left(ErrorResponse(NOT_FOUND, "Subscription Summary not found")))
+      case Some(subscriptionSummary) => getObligationAndFinancialInfo(alcoholDutyReference, subscriptionSummary).value
+    }
+  }
+  private def getObligationAndFinancialInfo(alcoholDutyReference: String, subscriptionSummary: SubscriptionSummary)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, ErrorResponse, AlcoholDutyCardData]                             =
+    if (subscriptionSummary.insolvencyFlag) {
+      EitherT.rightT(InsolventCardData(alcoholDutyReference))
```

Comment:
Move these logic to the getAlcoholDutyCardData, only leave the case where we do obligations and financials calls